apps/prairielearn/src/sync/syncFromDisk.ts

```diff
@@ -108,71 +105,33 @@ export async function syncDiskToSqlWithLock(courseId, courseDir, logger) {
   };
 }
 
-/**
- * @param {string} course_id
- * @param {string} courseDir
- * @param {any} logger
- * @param {(err: Error | null, result: SyncResults) => void} callback
- */
-function _syncDiskToSqlWithLock(course_id, courseDir, logger, callback) {
-  callbackify(async () => {
-    return await syncDiskToSqlWithLock(course_id, courseDir, logger);
-  })(callback);
-}
-
-/**
- * @param {string} course_id
- * @param {string} courseDir
- * @param {any} logger
- * @param {(err: Error | null, result?: SyncResults) => void} callback
- */
-function syncDiskToSql(course_id, courseDir, logger, callback) {
+export async function syncDiskToSql(
+  course_id: string,
+  courseDir: string,
+  logger: Logger,
+): Promise<SyncResults> {
   const lockName = getLockNameForCoursePath(courseDir);
   logger.verbose(chalkDim(`Trying lock ${lockName}`));
-  namedLocks.tryLock(lockName, (err, lock) => {
-    if (ERR(err, callback)) return;
-    if (lock == null) {
-      logger.verbose(chalk.red(`Did not acquire lock ${lockName}`));
-      callback(new Error(`Another user is already syncing or modifying the course: ${courseDir}`));
-    } else {
-      logger.verbose(chalkDim(`Acquired lock ${lockName}`));
-      _syncDiskToSqlWithLock(course_id, courseDir, logger, (err, result) => {
-        namedLocks.releaseLock(lock, (lockErr) => {
-          if (ERR(lockErr, callback)) return;
-          if (ERR(err, callback)) return;
-          logger.verbose(chalkDim(`Released lock ${lockName}`));
-          callback(null, result);
-        });
-      });
-    }
-  });
-}
-
-/**
- * @param {string} courseDir
- * @param {string} course_id
- * @param {any} logger
- * @returns {Promise<SyncResults>}
- */
-export function syncDiskToSqlAsync(course_id, courseDir, logger) {
-  // @ts-expect-error -- The types of `syncDiskToSql` can't express the fact
-  // that it'll always result in a non-undefined value if it doesn't error.
-  return promisify(syncDiskToSql)(course_id, courseDir, logger);
+  const lock = await namedLocks.tryLockAsync(lockName);
+  if (lock == null) {
+    logger.verbose(chalk.red(`Did not acquire lock ${lockName}`));
+    throw new Error(`Another user is already syncing or modifying the course: ${courseDir}`);
+  }
+  let result: SyncResults;
+  try {
+    logger.verbose(chalkDim(`Acquired lock ${lockName}`));
+    result = await syncDiskToSqlWithLock(course_id, courseDir, logger);
+  } finally {
+    await namedLocks.releaseLockAsync(lock);
+    logger.verbose(chalkDim(`Released lock ${lockName}`));
+  }
+  return result;
```

Comment:
Rather than using manual lock management, consider using `doWithLock` instead. See `lib/group-update.js` for examples.